reward/staking_manager.go

```diff
@@ -195,6 +213,71 @@ func updateStakingInfo(blockNum uint64) (*StakingInfo, error) {
 	return stakingInfo, nil
 }
 
+// NOTE: Even if the AddressBook contract code is erroneous and it returns unexpected result, this function should not return error in order not to stop block proposal.
+// getStakingInfoFromAddressBook returns stakingInfo fetched from AddressBook contract
+// 1. If calling AddressBook contract fails, it returns error
+// 2. If AddressBook is not activated, emptyStakingInfo is returned without error
+// 3. If AddressBook is activated, it returns fetched stakingInfo
+func getStakingInfoFromAddressBook(blockNum uint64) (*StakingInfo, error) {
+	caller := backends.NewBlockchainContractCaller(stakingManager.blockchain)
+	contract, _ := contract.NewAddressBookCaller(addressBookContractAddress, caller)
+
+	types, addrs, err := contract.GetAllAddress(&bind.CallOpts{BlockNumber: new(big.Int).SetUint64(blockNum)})
+	if err != nil {
+		return nil, fmt.Errorf("failed to call AddressBook contract. root err: %s", err)
+	}
+
+	if len(types) == 0 && len(addrs) == 0 {
+		// This is an expected behavior when the addressBook contract is not activated yet.
+		logger.Info("The addressBook is not yet activated. Use empty stakingInfo", "reason", err)
+		return newEmptyStakingInfo(blockNum), nil
+	}
+
+	if len(types) != len(addrs) {
+		logger.Error("Fail while parsing a result from the addressBook. Use empty staking info", "err", fmt.Errorf("length of type list and address list differ. len(type)=%d, len(addrs)=%d", len(types), len(addrs)))
+		return newEmptyStakingInfo(blockNum), nil
+	}
+
+	var (
+		nodeIds      = []common.Address{}
+		stakingAddrs = []common.Address{}
+		rewardAddrs  = []common.Address{}
+		pocAddr      = common.Address{}
+		kirAddr      = common.Address{}
+	)
+
+	// Parse and construct node information
+	for i, addrType := range types {
+		switch addrType {
+		case addressTypeNodeID:
+			nodeIds = append(nodeIds, addrs[i])
+		case addressTypeStakingAddr:
+			stakingAddrs = append(stakingAddrs, addrs[i])
+		case addressTypeRewardAddr:
+			rewardAddrs = append(rewardAddrs, addrs[i])
+		case addressTypePoCAddr:
+			pocAddr = addrs[i]
+		case addressTypeKIRAddr:
+			kirAddr = addrs[i]
+		default:
+			logger.Error("Fail while parsing a result from the addressBook. Use empty staking info", "err", fmt.Errorf("invalid type from AddressBook: %d", addrType))
+			return newEmptyStakingInfo(blockNum), nil
+		}
+	}
+
+	// validate parsed node information
+	if len(nodeIds) != len(stakingAddrs) ||
+		len(nodeIds) != len(rewardAddrs) ||
+		common.EmptyAddress(pocAddr) ||
+		common.EmptyAddress(kirAddr) {
+		// This is an expected behavior when the addressBook contract is not activated yet.
+		logger.Info("The addressBook is not yet activated. Use empty stakingInfo", "reason", err)
```

Comment:
No `err` exist for this context. And, this code is not reachable if AddressBook is not activated. Don't we need to return an error for this case?

Reply:
https://github.com/klaytn/klaytn/blob/2dc0b6578b6def6e80238351397170e92d747b0d/reward/addressbook_connector.go#L155-L160

Reply:
In the previous code, an error log was printed and `newEmptyStakingInfo(blockNum)` was returned

Reply:
@blukat29 handled this in #1882